apps/dotcdn/src/app/app.component.ts

```diff
@@ -0,0 +1,105 @@
+import { Component, OnInit, ViewChild } from '@angular/core';
+import { DotCDNState } from './app.interface';
+import { ChartPeriod } from './app.enums';
+import { FormGroup, FormBuilder } from '@angular/forms';
+import { DotCDNStore } from './dotcdn.component.store';
+import { Observable } from 'rxjs';
+import { SelectItem } from 'primeng/api';
+import { ChartOptions } from 'chart.js';
+import { take } from 'rxjs/operators';
+
+@Component({
+    selector: 'dotcms-root',
+    templateUrl: './app.component.html',
+    styleUrls: ['./app.component.scss']
+})
+export class AppComponent implements OnInit {
+    @ViewChild('chart', { static: true }) chart: any;
+    purgeZoneForm: FormGroup;
+    periodValues: SelectItem[] = [
+        { label: 'Last 30 days', value: ChartPeriod.Last30Days },
+        { label: 'Last 60 days', value: ChartPeriod.Last60Days }
+    ];
+
+    selectedPeriod: SelectItem<string> = { value: ChartPeriod.Last30Days };
+    vm$: Observable<Partial<DotCDNState>> = this.dotCdnStore.vm$;
```

Comment:
You need to type this correctly because using partial is not telling what this is, check the IDE, is have all the properties when IRL only have 3.

![image](https://user-images.githubusercontent.com/751424/115887335-0fb64400-a40f-11eb-9c2f-a6a1746b662c.png)